snapshot/snapshot.go

```diff
@@ -0,0 +1,529 @@
+package snapshot
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"time"
+)
+
+// manifestName is the name of the manifest file in the snapshot.
+const manifestName = "manifest"
+
+// Manifest represents a list of files in a snapshot.
+type Manifest struct {
+	Files []File `json:"files"`
+}
+
+// Diff returns a Manifest of files that are newer in s than other.
+func (m *Manifest) Diff(other *Manifest) *Manifest {
+	diff := &Manifest{}
+
+	// Find versions of files that are newer in m.
+loop:
+	for _, a := range m.Files {
+		// Try to find a newer version of the file in other.
+		// If found then don't append this file and move to the next file.
+		for _, b := range other.Files {
+			if a.Name != b.Name {
+				continue
+			} else if !a.ModTime.After(b.ModTime) {
+				continue loop
+			} else {
+				break
+			}
+		}
+
+		// Append the newest version.
+		diff.Files = append(diff.Files, a)
+	}
+
+	// Sort files.
+	sort.Sort(Files(diff.Files))
+
+	return diff
+}
+
+// Merge returns a Manifest that combines s with other.
```

Comment:
Fixed: 782231b, it used to be called `Snapshot` so it's variable was `s`. `Snapshot` is now `Manifest` since that makes more sense.